azext_iot/tests/iothub/devices/test_iot_edge_devices_unit.py

```diff
@@ -786,6 +813,14 @@ def test_create_edge_device_config(
             device_config.hostname if device_config.hostname else "{{HOSTNAME}}"
         )
 
+        # respect existing config reprovisioning mode
+        if device_config_path:
+            with open(device_config_path, "rb") as file:
+                expected_config = tomli.load(file)
+                assert device_toml["auto_reprovisioning_mode"] == getattr(expected_config, "auto_reprovisioning_mode", "Dynamic")
```

Comment:
The test case with the device_config_path is pointing to [this one](https://github.com/Azure/azure-iot-cli-extension/blob/dev/azext_iot/tests/iothub/devices/device_configs/device_config.toml) , right? In that case when is auto_reprovisioning_mode not Dynamic?  I don't see the `auto_reprovisioning_mode` field set in the file.

Reply:
It's actually changed in this PR - [I've set the test `config.toml` to use `auto_reprovisioning_mode = "OnErrorOnly"`](https://github.com/Azure/azure-iot-cli-extension/pull/613/files#diff-7ca392c8d88a5cc372d64ce3e35b9ede99a5ef0f7951edd65f264e9bf4808cf1R6)

Reply:
And just for clarity - this test file is not the one used as the basis / default template if one isn't provided, that's all kept in  the `edge_device_config.py` [here](https://github.com/Azure/azure-iot-cli-extension/pull/613/files#diff-b9fea65352456a520c7b7f54f2dd74b165bf4b1120b7f12e07f5749a0c1807e9R106)

Reply:
Missed that thanks....do we need the "Dynamic" fallback in `getattr`?

Reply:
I believe so - I want to check that:
- If there is an existing mode provided in the config, ensure that is in the resultant config
- If there isn't, ensure it gets set to "Dynamic"

To be more explicit, I could change it to:
```suggestion
if not getattr(expected_config, "auto_reprovisioning_mode", None):
    assert device_toml["auto_reprovisioning_mode"] == "Dynamic"
else:
    assert device_toml["auto_reprovisioning_mode"] == expected_config["auto_provisioning_mode"]
```

I see what you mean, it kind of hides (even if the result is the same) whether the expected config had no value or a value of "Dynamic"

Reply:
Yea that's what I was getting at, potentially impacting the intent of the test, I like the more explicit refactor